### PR TITLE
Use dns-java 3.0 which adds support for dns ttl

### PR DESCRIPTION
**Pull request by spkrka**
Also remove the option of setting dns refresh period
manually.

### CODE REVIEW

**Comment by protocol7** [comment]
+1

File: src/main/java/com/spotify/folsom/ketama/SrvKetamaClient.java [hunk 1]
```diff
@@ -99,17 +98,33 @@ public void updateDNS() {
...
+                .sortedCopy(hosts);
+        if (!newAddresses.equals(addresses)) {
+          addresses = newAddresses;
+          log.info("Connecting to " + newAddresses);
+          List<AddressAndClient> addressAndClients = getAddressesAndClients(newAddresses);
+          setPendingClient(addressAndClients);
+        }
+      } finally {
+        long delay = clamp(MIN_DNS_WAIT_TIME, MAX_DNS_WAIT_TIME, ttl);
+        refreshJob = this.executor.schedule(refreshRunnable, delay, TimeUnit.SECONDS);
```
Comment (danielnorberg) [inline src/main/java/com/spotify/folsom/ketama/SrvKetamaClient.java:124]: Seems we'd wait 1 hour before attempting to resolve again if there was no records found. That'll happen from time to time in prod due to infra breakage.

How about a special case where we retry earlier when no records were found?

Comment (spkrka) [inline src/main/java/com/spotify/folsom/ketama/SrvKetamaClient.java:124]: True, didn't think about that case. Maybe we should keep the old setting then, but have it be treated as either the max value or the default if there are no hosts found.

Comment (spkrka) [inline src/main/java/com/spotify/folsom/ketama/SrvKetamaClient.java:124]: fixed now with a force push.

**Comment by danielnorberg** [comment]
:+1:

**Pull request merged**